compliance/index/v4/node_test.go

```diff
@@ -0,0 +1,155 @@
+package v4
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/quay/claircore"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestNodeIndexerSuite(t *testing.T) {
+	suite.Run(t, new(nodeIndexerSuite))
+}
+
+type nodeIndexerSuite struct {
+	suite.Suite
+}
+
+func createLayer(path string) (layer *claircore.Layer, e error) {
+	testdir, err := filepath.Abs(path)
+	if err != nil {
+		return nil, err
+	}
+	layer, err = constructLayer(context.TODO(), layerDigest, testdir)
+	if err != nil {
+		return nil, err
+	}
+	return layer, nil
+}
+
+func (s *nodeIndexerSuite) TestConstructLayer() {
+	testdir, err := filepath.Abs("testdata")
+	s.NoError(err)
+
+	layer, err := constructLayer(context.TODO(), layerDigest, testdir)
+	defer func() {
+		if tmpErr := layer.Close(); tmpErr != nil {
+			err = tmpErr
+		}
+	}()
+
+	s.NoError(err)
+	s.NotNil(layer)
+}
+
+func (s *nodeIndexerSuite) TestConstructLayerNoURI() {
+	layer, err := constructLayer(context.TODO(), layerDigest, "")
+	defer func() {
+		if tmpErr := layer.Close(); tmpErr != nil {
+			err = tmpErr
+		}
+	}()
+	s.ErrorContains(err, "no URI provided")
+}
+
+func (s *nodeIndexerSuite) TestConstructLayerIllegalDigest() {
+	layer, err := constructLayer(context.TODO(), "sha256:nodigest", s.T().TempDir())
+	defer func() {
+		if tmpErr := layer.Close(); tmpErr != nil {
+			err = tmpErr
+		}
+	}()
```

Comment:
I think that deferred statement like this in test has no value - it will be called only after the last assert from that test.

Reply:
See my other comment about error checking of deferred function calls. 
IIUC, this allows us to still fail the test when layer.Close errors out.
Also, we do get a linter failure if the errors are left unchecked, be it in the test or the cod itself